examples/py_triple.rs

```diff
@@ -0,0 +1,411 @@
+                config.s_mul.enable(&mut region, row_offset)?;
+
+                let l_cell = region.assign_advice(
+                    || "l",
+                    config.l_col,
+                    row_offset,
+                    || Ok(prev_alloc.value),
+                )?;
+                let r_cell = region.assign_advice(
+                    || "r",
+                    config.r_col,
+                    row_offset,
+                    || Ok(prev_alloc.value),
+                )?;
+                region.constrain_equal(&config.perm, prev_alloc.cell, l_cell)?;
+                region.constrain_equal(&config.perm, prev_alloc.cell, r_cell)?;
+
+                let o_cell = region.assign_advice(
+                    || "l * r",
+                    config.o_col,
+                    row_offset,
+                    || Ok(squared_value),
+                )?;
+
+                squared_alloc = Some(Alloc { cell: o_cell, value: squared_value });
+                Ok(())
+            },
+        )?;
+
+        Ok(squared_alloc.unwrap())
+    }
+
+    // In the next available row, copies the previously allocated values `l_prev_alloc`, `r_prev_alloc`,
+    // and `o_prev_alloc` into the row's left, right, and output cells respectively. Enabling the
+    // `s_add` selector enforces that the values written in the row satisfy the addition constraint
+    // `l + r = o`.
+    // 
+    // This function is called `constrained_add` because the output of `l + r` is provided by the
+    // function caller as a previously allocated value.
+    fn constrained_add(
+        &self,
+        mut layouter: impl Layouter<Fp>,
+        l_in_alloc: Alloc,
+        r_in_alloc: Alloc,
+        o_in_alloc: Alloc,
+    ) -> Result<(), Error> {
+        let config = self.config();
+
+        layouter.assign_region(
+            || "constrained add",
+            |mut region: Region<'_, Fp>| {
+                let row_offset = 0;
+                config.s_add.enable(&mut region, row_offset)?;
+
+                let l_cell = region.assign_advice(
+                    || "l",
+                    config.l_col,
+                    row_offset,
+                    || Ok(l_in_alloc.value),
+                )?;
+                let r_cell = region.assign_advice(
+                    || "r",
+                    config.r_col,
+                    row_offset,
+                    || Ok(r_in_alloc.value),
+                )?;
+                let o_cell = region.assign_advice(
+                    || "o",
+                    config.o_col,
+                    row_offset,
+                    || Ok(o_in_alloc.value),
+                )?;
+
+                region.constrain_equal(&config.perm, l_in_alloc.cell, l_cell)?;
+                region.constrain_equal(&config.perm, r_in_alloc.cell, r_cell)?;
+                region.constrain_equal(&config.perm, o_in_alloc.cell, o_cell)?;
+
+                Ok(())
+            },
+        )?;
+
+        Ok(())
```

Comment:
Since the region already returns `Ok(())`:
```suggestion
        )
```